File: adventofcode/go/internal/year2015/day15/day15_test.go
package day15

import (
	"testing"

	"github.com/Saser/pdp/adventofcode/go/internal/testcase"
)

var (
	exampleFile = testcase.Runfile("adventofcode/go/internal/year2015/day15/testdata/example")
	inputFile   = testcase.Runfile("adventofcode/inputs/2015/15")

	tcPart1 = testcase.NewFile("input", inputFile, "13882464")
	tcPart2 = testcase.NewFile("input", inputFile, "11171160")
)

func TestPart1(t *testing.T) {
	for _, tc := range []testcase.TestCase{
		testcase.NewFile(exampleFile, exampleFile, "62842880"),
		tcPart1,
	} {
		tc.Test(t, Part1)
	}
}

func BenchmarkPart1(b *testing.B) {
	tcPart1.Benchmark(b, Part1)
}

func TestPart2(t *testing.T) {
	for _, tc := range []testcase.TestCase{
		testcase.NewFile(exampleFile, exampleFile, "57600000"),
		tcPart2,
	} {
		tc.Test(t, Part2)
	}
}

func BenchmarkPart2(b *testing.B) {
	tcPart2.Benchmark(b, Part2)
}

File: adventofcode/rust/year2017/day15/day15_test.rs
use adventofcode_rust_aoc as aoc;

mod part1 {
    use super::*;
    aoc::testfn!(
        example,
        aoc::Data::File("adventofcode/rust/year2017/testdata/day15/ex"),
        aoc::Data::String("588"),
        crate::part1
    );
    aoc::testfn!(
        actual,
        aoc::Data::File("adventofcode/inputs/2017/15"),
        aoc::Data::String("609"),
        crate::part1
    );
}

mod part2 {
    use super::*;
    aoc::testfn!(
        example,
        aoc::Data::File("adventofcode/rust/year2017/testdata/day15/ex"),
        aoc::Data::String("309"),
        crate::part2
    );
    aoc::testfn!(
        actual,
        aoc::Data::File("adventofcode/inputs/2017/15"),
        aoc::Data::String("253"),
        crate::part2
    );
}
